src/Drupal/DrupalExtension/Context/RawDrupalContext.php

```diff
@@ -476,20 +529,23 @@ public function login() {
     $submit->click();
 
     if (!$this->loggedIn()) {
-      if (isset($this->user->role)) {
-        throw new \Exception(sprintf("Unable to determine if logged in because 'log_out' link cannot be found for user '%s' with role '%s'", $this->user->name, $this->user->role));
+      if (isset($user->role)) {
+        throw new \Exception(sprintf("Unable to determine if logged in because 'log_out' link cannot be found for user '%s' with role '%s'", $user->name, $user->role));
       }
       else {
-        throw new \Exception(sprintf("Unable to determine if logged in because 'log_out' link cannot be found for user '%s'", $this->user->name));
+        throw new \Exception(sprintf("Unable to determine if logged in because 'log_out' link cannot be found for user '%s'", $user->name));
       }
     }
+
+    $manager->setCurrentUser($user);
   }
 
   /**
    * Logs the current user out.
    */
   public function logout() {
     $this->getSession()->visit($this->locatePath('/user/logout'));
+    $this->getUserManager()->setCurrentUser(FALSE);
```

Comment:
I added this change because this seems to have been missing before. If we log the user out through the UI we should also clear the current user.